### PR TITLE
Feature/3219 iap analytics2

### LINKED ISSUE #3219: Set up Analytics for IAP
# Description

Upgrade the existing GA account from v3 to v4 and apply to the app as well as TC-report.
Share analytics about the IAP page with IAP team.

## Background.

The IAP team has requested the ability to track page views. 

First, we should check if Azure already provides traffic analytics from which we could create reports for them.

If that is not possible, we should [set up a Google Analytics account](https://support.google.com/analytics/answer/10759417). From a cursory look at the [account structure](https://support.google.com/analytics/answer/9679158), we may want to create a separate "property" for IAP.

We will want to configure this to collect minimal data, and then add the necessary script to the IAP landing page.

In the future, we will likely want to set this up on all our pages (probably under different "properties" if using Google Analytics), but we can start with a minimal implementation, just on the IAP homepage.

### PR DESCRIPTION
**Pull request by brindasasi**
REsolves #3219 
Since the below branch became stale before Hydrogen merge .. it was seriously broken with dependencies and had to create  this pr now. No functionality changed. 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/3675 

### How to Test Local
Check out this branch
Add "InstrumentationKey=8f4f1fd8-8831-4a3b-84a0-20245c86e418;IngestionEndpoint=https://canadacentral-1.in.applicationinsights.azure.com/;LiveEndpoint=https://canadacentral.livediagnostics.monitor.azure.com/" as the value for APPLICATIONINSIGHTS_CONNECTION_STRING in .env
Open the network tab in your browser
Navigate to '/en/indigenous-it-apprentice'
Search for track in the network tab and verify the headers and payload 

### How to verify the metrics in  Azure 

- Deploy the branch in dev 
- Add the env variable a above 
- Navigate to IAP home page and make some requests
- login to chrome with your pc account and go to portal.azure.com
- click "AppInsights"
- click "Transaction Search" and look for the hits for the time you hit the page 
image (3)

### Deployment Notes

**Dev & UAT**
Add the below environment variable
APPLICATIONINSIGHTS_CONNECTION_STRING=""

( empty string as we don't want to collect metrics from these environments )

**Prod**
APPLICATIONINSIGHTS_CONNECTION_STRING= "InstrumentationKey=8f4f1fd8-8831-4a3b-84a0-20245c86e418;IngestionEndpoint=https://canadacentral-1.in.applicationinsights.azure.com/;LiveEndpoint=https://canadacentral.livediagnostics.monitor.azure.com/"

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,3 +1,6 @@
 {
-  "note": "Workaround. Bundlewatch CLI needs package.json in root."
+  "note": "Workaround. Bundlewatch CLI needs package.json in root.",
+  "dependencies": {
```
Comment (mnigh) [inline package.json:3]: The package `applicationinsights` is not used and can be removed.

**Review by mnigh — DISMISSED** [review]
A few things to clean up, but after making those changes locally, I do see the tracking beacon with a payload being sent:

### Screenshot
Screen Shot 2022-09-13 at 10 14 06

**Comment by petertgiles** [comment]
Very cool!  I can see the tracker messages and was able to find my activity in the Azure Portal.  When I drilled into the transaction details I could see that it located me down to my city!  But this data isn't really usable by our clients yet.  We need a dashboard or a report that can be sent to them, I think.

**Comment by brindasasi** [comment]
> Very cool! I can see the tracker messages and was able to find my activity in the Azure Portal. When I drilled into the transaction details I could see that it located me down to my city! But this data isn't really usable by our clients yet. We need a dashboard or a report that can be sent to them, I think.

Oh I did create a dashboard .. forgot to share it here.. on the AppInsights homepage click on "Application Dashboard" . It will bring up all the charts.. but there are too many metrics rt now. We need to know what the user really wants to know in order to customize this. That could go on a sep issue. 

Sharing options for the Dashboard 

- Providing access control for the elite users to actually come & see the dashboard in azure
- Export to pdf , download as json is available. 


-

**Comment by brindasasi** [comment]
> > Oh I did create a dashboard
> 
> Oh, nice, I see it. Could you put a geolocation map on it? We know for sure that's one thing they want.

Added a widget of bar chart with unique users by City. Does that work?

**Comment by petertgiles** [comment]
> Added a widget of bar chart with unique users by City. Could you check?

Eh, a bar chart is a bit weak.  :smile: Could you get a sample over to the IAP team to see if this dashboard works for them?

**Comment by brindasasi** [comment]
> > Added a widget of bar chart with unique users by City. Could you check?
> 
> Eh, a bar chart is a bit weak. 😄 Could you get a sample over to the IAP team to see if this dashboard works for them?

Shared with IAP team and they seems to be ok with it.

**Review by petertgiles — APPROVED** [review]
> Shared with IAP team and they seems to be ok with it.

Oh nice!  :partying_face:

**Pull request merged**